app/livechat/server/hooks/saveAnalyticsData.ts

```diff
@@ -1,17 +1,44 @@
 import { callbacks } from '../../../../lib/callbacks';
-import { LivechatRooms } from '../../../models';
+import { LivechatRooms } from '../../../models/server/index';
 import { normalizeMessageFileUpload } from '../../../utils/server/functions/normalizeMessageFileUpload';
 
+type MessageData = {
+	editedAt: Date;
+	token: string;
+	file: string;
+};
+
+type RoomData = {
+	t: string;
+	metrics: {
+		v: {
+			lq: string;
+		};
+		servedBy: {
+			ts: string;
+			lr: string;
+		};
+		response: {
+			total: number;
+			tt: number;
+		};
+	};
+	ts: string;
+	servedBy: {
+		ts: string;
+	};
+};
+
 callbacks.add(
 	'afterSaveMessage',
-	function (message, room) {
+	function (message: MessageData, room: RoomData | undefined): MessageData {
```

Comment:
Refer to prev comments. Most of those props should exist on IOmnichannelRoom interface 👀